sql-plugin/src/main/scala/com/nvidia/spark/rapids/ParquetCachedBatchSerializer.scala

```diff
@@ -1096,14 +1096,17 @@ protected class ParquetCachedBatchSerializer extends GpuCachedBatchSerializer wi
      * is called on this iterator
      */
     class ColumnarBatchToCachedBatchIterator extends InternalRowToCachedBatchIterator {
+      // Exposing this just for tests to close the batch
+      var hostBatch: ColumnarBatch = null
```

Comment:
Not necessarily for this PR, but should `hostBatch` be closed proactively when the iterator is exhausted?  We don't want to unnecessarily hold onto host batches until the task finishes.

Reply:
We are mapping the GPU column to HOST column and that happens lazily. Even though the iterator is exhausted, the hostBatch is still needed later downstream which is why we are using the `TaskContext`.

Reply:
But eventually downstream will iterate past the hostBatch, and then the hostBatch should close.  Or am I missing some place where it is still needed?  Curious why we need to wait until the entire task completes to know it's safe to close the hostBatch.  We should not rely on the task completion to close the host batch in the typical case where everything is consumed.